core/src/lib/bsys.cc

```diff
@@ -233,6 +233,22 @@ char* bstrinlinecpy(char* dest, const char* src)
  */
 char* bstrncpy(char* dest, const char* src, int maxlen)
 {
+  std::string tmp;
+
+  /*
+  if (dest <= src) {
+    Dmsg6(100, "dest=%p, src=%p, len=%u, size=%u, len*s=%u, dest+size=%p\n",
+  dest, src, maxlen, sizeof(char),  maxlen * sizeof(char), dest + maxlen *
+  sizeof(char)); ASSERT((dest + (maxlen - 1) * sizeof(char)) < src)
+  }
+  */
+
+  if ((dest <= src) && ((dest + (maxlen - 1) * sizeof(char)) >= src)) {
```

Comment:
Should implement a unittest for this.

Reply:
Makes sense.

Reply:
I already implemented `ConfigParser_Dir.CFG_TYPE_TIME` to test this. But I assume, you mean a more specific test for only this function. Okay, I'll do that.

Reply:
Added unittest for bstrncpy.